rmw_zenoh_cpp/src/detail/rmw_data_types.cpp

```diff
@@ -102,7 +104,7 @@ std::unique_ptr<saved_msg_data> rmw_subscription_data_t::pop_next_message()
     return nullptr;
   }
 
-  std::unique_ptr<saved_msg_data> msg_data = std::move(message_queue_.front());
+  std::unique_ptr<rmw_zenoh_cpp::saved_msg_data> msg_data = std::move(message_queue_.front());
```

Comment:
This change shouldn't be necessary.

(same goes for all of the changes within this namespace below)